docs/strategies/circuit-breaker.md

```diff
@@ -90,6 +90,197 @@ await manualControl.CloseAsync();
 | `ManualControl`     | `null`                                                                     | Allows for manual control to isolate or close the circuit.                                 |
 | `StateProvider`     | `null`                                                                     | Enables the retrieval of the current state of the circuit.                                 |
 
+## Diagrams
+
+### State diagram
+
+```mermaid
+%%{init: {'theme':'dark'}}%%
+stateDiagram-v2
+direction LR
+    [*] --> Closed
+    Closed --> Open: Exceeds threshold
+    Open --> HalfOpen: Elapses break duration
+    HalfOpen --> Closed: Passes the probe
+    HalfOpen --> Open: Fails the probe
+```
+
+Whenever someone says _the circuit breaks_ that means CB transitions from `Closed` to `Open` state.
+
+### Simple
+
+Let's suppose we have a circuit breaker strategy with `SamplingDuration`: `2 seconds`, `MinimumThroughput`: `2` and `FailureRatio` : `0.5`.
+
+#### Simple: happy path sequence diagram
+
+The circuit will not break because the actual failure ratio (0.33) will be below the threshold (0.5) after the 3rd call.
+
+```mermaid
+%%{init: {'theme':'dark'}}%%
+sequenceDiagram
+    autonumber
+    actor C as Caller
+    participant P as Pipeline
+    participant CB as CircuitBreaker
+    participant D as DecoratedUserCallback
+
+    C->>P: Calls ExecuteAsync
+    P->>CB: Calls ExecuteCore
+    Note over CB: Closed state
+    Note over CB, D: Sampling start
```

Comment:
Might make sense to "activate" the line here to show the sampling being active in these diagrams.